.github/workflows/build.yml

```diff
@@ -69,24 +69,44 @@ jobs:
     if: ${{ needs.check-changes.outputs.has_changes == 'yes' || github.event_name == 'push' }}
     runs-on: [ubuntu-latest]
     steps:
+    - name: Free disk space
+      # https://github.com/actions/virtual-environments/issues/709
+      run: |
+        sudo apt-get clean
+        df -h
     - uses: actions/checkout@v4
       with:
         show-progress: false
+    - uses: actions/setup-go@v4
+      with:
+        go-version-file: 'go.mod'
     - name: Build Antrea UBI8 Docker image without pushing to registry
       if: ${{ github.repository != 'antrea-io/antrea' || github.event_name != 'push' || github.ref != 'refs/heads/main' }}
       run: |
         ./hack/build-antrea-linux-all.sh --pull --distro ubi
-    - name: Build and push Antrea UBI8 Docker image to registry
-      if: ${{ github.repository == 'antrea-io/antrea' && github.event_name == 'push' && github.ref == 'refs/heads/main' }}
-      env:
-        DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
-        DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+    - name: Install Kind
```

Comment:
I also don't think this is elegant. Should we move the test part out as a separate job?

Reply:
How about we make it daily/bidaily job to avoid adding extra time to very basic jobs? I'm more concerned more tests would be added like this if there is a precedent. We could potentially move more rarely failed but long running tests to daily/bidaily job.

Reply:
In the original issue (#5729) I suggested adding a very basic job just to make sure that the UBI-based image is not broken in a pretty obvious way (like it was around 1.14). I was hoping impact on PR test time would be minimal.
One issue that we have today is that even if someone modifies the UBI Dockerfile, we do not check anything besides making sure that the image still builds.

I am not a big fan of the daily jobs, as they can be broken for a long time before someone actually starts looking into the failure.

I think if we go with my suggestion above: skip building the UBI image in `build.yml` for PRs, and build the image + sanity check it in `kind_ubi.yml`, the impact on test time will be minimal. The sanity check itself should only take a couple of minutes. Additionally, after I fix https://github.com/antrea-io/antrea/issues/5944, we should see build time reduction for most jobs.

I also think that we should add a trigger to `conformance.yml` so that we run the K8s conformance tests (at least for the UBI image) for every _release_ PR.